swift-sdk/Internal/EmbeddedSessionManager.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  EmbeddedSessionManager.swift
+//  
+//
+//  Created by Andrew Nelson on 5/30/23.
+//
+
+import Foundation
+
+public class EmbeddedSessionManager {
+    public static let shared = EmbeddedSessionManager()
+    public var currentSession: IterableEmbeddedSession?
+    var currentlyTrackingImpressions: [String: (currentDisplayDuration: TimeInterval, startTime: Date, tracking: Bool)] = [:]
```

Comment:
`totalDisplayDuration` instead of `currentDisplayDuration` as we are accumulating the total over time.